src/sync/impl/sync_client.hpp

```diff
@@ -33,11 +35,23 @@ struct SyncClient {
 
     SyncClient(std::unique_ptr<util::Logger> logger,
                ReconnectMode reconnect_mode = ReconnectMode::normal,
-               bool verify_ssl = true)
+               bool verify_ssl = true,
+               BindingCallbackThreadObserver *client_thread_listener = nullptr)
```

Comment:
Why take this as an argument rather than just having the lambda below use the global variable?